src/components/input/gamepadtokey.js

```diff
@@ -347,10 +347,54 @@ require(['apphost'], function (appHost) {
             }
         }
         // Schedule the next one
-        requestAnimationFrame(runInputLoop);
+        inputLoopTimer = requestAnimationFrame(runInputLoop);
     }
 
-    runInputLoop();
+    function startInputLoop() {
+        if (!inputLoopTimer) {
+            runInputLoop();
+        }
+    }
+
+    function stopInputLoop() {
+        cancelAnimationFrame(inputLoopTimer);
+        inputLoopTimer = undefined;
+    }
+
+    function isGamepadConnected() {
+        var gamepads = navigator.getGamepads();
+        for (var i = 0, len = gamepads.length; i < len; i++) {
+            var gamepad = gamepads[i];
+            if (gamepad && gamepad.connected) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function attachGamepad(e) {
```

Comment:
```suggestion
    function attachGamepadOrFocus(e) {
```
or something similar, so we don't confuse future devs on why `attachGamepad` is called upon focusing a page...

Reply:
@JustAMan I suppose you mean ``attachGamepadOnFocus``, right?

Reply:
No, he means that same handler is attached to `attach` and `focus` and `detach` and `blur`. So name of function and its use may confuse devs.

Reply:
As option for names:
`attachGamepad` -> `tryStartInputLoop`
`detachGamepad` (_with single 't'_) -> `tryStopInputLoop`

On the other hand, `attachGamepad` could be treated as "attach gamepad to document".

Reply:
@dmitrylyzo Yeah I know, but ``attachGamepadOrFocus`` doesn't make any sense: with that name seems that the function or attach the gamepad, or focus the page, which is not true at all.
The function has an ``and`` logic, meaning that it runs when a gamepad exists **and** the document is focused. That's why I think that @JustAMan meant ``attachGamepadOnFocus``, because it's executed as long as the document is focused.

About the other names, let's see @JustAMan opinion's.

Reply:
I think it's fine for now, this is a vast improvement over the current behavior.

Reply:
Maybe then `onFocusOrGamepadAttach`?..

Reply:
@JustAMan But why the ``or``? Both should happen at the same time for i to be run 😅😅.

Reply:
Because either of those events fires off the callback. It doesn't matter which conditions are expected inside, but rather _when_ it is called, IMHO.

Reply:
I understand your point now. I renamed to ``onFocusOrGamepadAttach`` and ``onFocusOrGamepadDetach``, which I think that it's really clear for upcoming devs who are used to expect to name things expecting the result of the conditions (like) and the events that runs the callback.